### PR TITLE
BRIDGE-1144 Fix crash when tapping on share

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**

### CODE REVIEW

File: APCAppCore/APCAppCore/UI/Onboarding/APCShareViewController.m [hunk 1]
```diff
@@ -307,7 +307,7 @@ - (void)messageComposeViewController:(MFMessageComposeViewController *)controlle
 
 - (IBAction) okayTapped: (id) __unused sender
 {
-    if (self.step != nil) {
+    if (self.taskViewController != nil) {
         // If this is a step view controller then dismiss
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/UI/Onboarding/APCShareViewController.m:311]: is this comment still apropos?

File: APCAppCore/APCAppCore/UI/Onboarding/EmailVerification/APCEmailVerifyViewController.m [hunk 2]
```diff
@@ -360,7 +360,7 @@ - (void) showThankYouPage
 {
     [self hideSpinnerUsingAnimation: YES andThenDoThis:^{
         
-        if (self.step) {
+        if (self.taskViewController != nil) {
             // If this has a step then it is part of a taskViewController
```
Comment (Erin-Mounts) [inline APCAppCore/APCAppCore/UI/Onboarding/EmailVerification/APCEmailVerifyViewController.m:364]: apropos comment still?

**Comment by Erin-Mounts** [comment]
couple of nitpicky comments about comments, otherwise looks good

**Comment by syoung-smallwisdom** [comment]
Yes, comments are applicable. Basically, in order to use these view controllers in a task flow then they have to inherit from `ORKStepViewController`, but that class checks the private `_step` ivar and asserts if it is nil. The hack-around is to have them inherit from a class that will add a dummy step if the step is nil. Because of that, they have to check for taskViewController being non-nil rather than the step being non-nil. (oops)

Ideally, this will all be stripped away, but not by Friday. :}

**Pull request merged**